### PR TITLE
Improve error handling for virtio-mem and virtio-iommu

### PR DESCRIPTION
**Pull request by likebreath**
virtio-devices: mem: Report error instead of panic
virtio-devices: iommu: Switch to use 'thiserror' 
virtio-devices: iommu: Propagate errors of processing request queue

Signed-off-by: Bo Chen <chen.bo@intel.com>

### CODE REVIEW

**Review by sboeuf — APPROVED** [review]

File: virtio-devices/src/iommu.rs [hunk 1]
```diff
@@ -282,76 +282,44 @@ unsafe impl ByteValued for VirtioIommuProbeProperty {}
 unsafe impl ByteValued for VirtioIommuProbeResvMem {}
 unsafe impl ByteValued for VirtioIommuFault {}
 
-#[derive(Debug)]
+#[derive(Error, Debug)]
 enum Error {
-    /// Guest gave us bad memory addresses.
+    #[error("Guest gave us bad memory addresses: {0}.")]
```
Comment (rbradford) [inline virtio-devices/src/iommu.rs:287]: Not sure the punctuation makes sense. Is this consistent with other error descriptions?

**Comment by rbradford** [comment]
(If you want to trivially create PR descriptions for multi-commit PRs then you can use `gh create`)

File: virtio-devices/src/iommu.rs (hunk 1 above)
Comment (rbradford) [reply virtio-devices/src/iommu.rs:287]: I looked through our code base and we're inconsistent. Definitely something to follow-up with later.

Comment (likebreath) [reply virtio-devices/src/iommu.rs:287]: @rbradford Do you think we should keep all error messages without period at the end, given they are (mostly) not complete sentences?

**Pull request merged**